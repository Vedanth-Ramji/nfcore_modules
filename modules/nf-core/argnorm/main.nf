process ARGNORM {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/argnorm:0.5.0--pyhdfd78af_0':
        'biocontainers/argnorm:0.5.0--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(input_tsv)
    val(db)
    val(hamronized)
    val(tool)

    output:
    tuple val(meta), path("*.tsv"), emit: tsv
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.5.0'
    """
    argnorm \\
        $tool \\
        -i $input_tsv \\
        -o $prefix \\
        --db $db \\
        $hamronized

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        argnorm: $VERSION
    END_VERSIONS
    """
}
