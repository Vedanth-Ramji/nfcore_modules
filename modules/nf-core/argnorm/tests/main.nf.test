nextflow_process {
    name "Test Process ARGNORM"
    script "../main.nf"
    process "ARGNORM"

    tag "modules"
    tag "modules_nfcore"
    tag "argnorm"

    test("argnorm - amrfinderplus_ncbi_raw - tsv") {
        config './argnorm_raw.config'

        when {
            process {
                """
                input[0] = [
                    [ id:'argnorm_raw.tsv' ], // meta map
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/amrfinderplus/test_output.tsv", checkIfExists: true)
                ]
                input[1] = 'amrfinderplus'
                input[2] = 'ncbi'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("argnorm - amrfinderplus_ncbi_hamronized - tsv") {
        config './argnorm_hamronized.config'

        when {
            process {
                """
                input[0] = [
                    [ id:'argnorm_hamronized.tsv' ], // meta map
                    file("https://raw.githubusercontent.com/BigDataBiology/argNorm/main/examples/hamronized/amrfinderplus.ncbi.orfs.tsv", checkIfExists: true)
                ]
                input[1] = 'amrfinderplus'
                input[2] = 'ncbi'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("argnorm - amrfinderplus_ncbi_hamronized - tsv - stub") {
        options "-stub"
        config './argnorm_hamronized.config'

        when {
            process {
                """
                input[0] = [
                    [ id:'argnorm_hamronized_stub.tsv'  ], // meta map
                    file("https://raw.githubusercontent.com/BigDataBiology/argNorm/main/examples/hamronized/amrfinderplus.ncbi.orfs.tsv", checkIfExists: true)
                ]
                input[1] = 'amrfinderplus'
                input[2] = 'ncbi'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("argnorm - amrfinderplus_ncbi - tsv - stub") {

        options "-stub"
        config './argnorm_raw.config'

        when {
            process {
                """
                input[0] = [
                    [ id:'argnorm_raw_stub.tsv', single_end:false ], // meta map
                    file("https://raw.githubusercontent.com/BigDataBiology/argNorm/main/examples/raw/amrfinderplus.ncbi.orfs.tsv", checkIfExists: true)
                ]
                input[1] = 'amrfinderplus'
                input[2] = 'ncbi'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
