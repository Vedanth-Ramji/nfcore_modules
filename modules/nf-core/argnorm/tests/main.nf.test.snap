{
    "argnorm - amrfinderplus_ncbi_raw - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.tsv",
                            "single_end": false
                        },
                        "test.tsv:md5,f870c239182592a065d9f80732b39bba"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e200075d98a6f59137f105efceea0426"
                ],
                "tsv": [
                    [
                        {
                            "id": "test.tsv",
                            "single_end": false
                        },
                        "test.tsv:md5,f870c239182592a065d9f80732b39bba"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e200075d98a6f59137f105efceea0426"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-02T14:56:50.742114812"
    },
    "argnorm - amrfinderplus_ncbi_hamronized - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.tsv",
                            "single_end": false
                        },
                        "test.tsv:md5,1f9a3820f09fd6a818af372dfe5cf322"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e200075d98a6f59137f105efceea0426"
                ],
                "tsv": [
                    [
                        {
                            "id": "test.tsv",
                            "single_end": false
                        },
                        "test.tsv:md5,1f9a3820f09fd6a818af372dfe5cf322"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e200075d98a6f59137f105efceea0426"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-02T14:57:19.694510786"
    },
    "argnorm - amrfinderplus_ncbi_hamronized - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.tsv",
                            "single_end": false
                        },
                        "test.tsv:md5,1f9a3820f09fd6a818af372dfe5cf322"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e200075d98a6f59137f105efceea0426"
                ],
                "tsv": [
                    [
                        {
                            "id": "test.tsv",
                            "single_end": false
                        },
                        "test.tsv:md5,1f9a3820f09fd6a818af372dfe5cf322"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e200075d98a6f59137f105efceea0426"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-02T14:57:47.422476341"
    },
    "argnorm - amrfinderplus_ncbi - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.tsv",
                            "single_end": false
                        },
                        "test.tsv:md5,e41b52c1213d8833c37472e3efb24db4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e200075d98a6f59137f105efceea0426"
                ],
                "tsv": [
                    [
                        {
                            "id": "test.tsv",
                            "single_end": false
                        },
                        "test.tsv:md5,e41b52c1213d8833c37472e3efb24db4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e200075d98a6f59137f105efceea0426"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-02T14:58:14.950164297"
    }
}